Multisig/UI/Settings/OwnerKeyManagement/WalletConnectOwnerKey/OnboardingConnectOwnerKeyViewController.swift

```diff
@@ -48,28 +48,43 @@ class OnboardingConnectOwnerKeyViewController: AddKeyOnboardingViewController {
             enterNameVC.address = address
             enterNameVC.badgeName = KeyType.walletConnect.imageName
 
-            enterNameVC.completion = { [unowned self] name in
+            enterNameVC.completion = { [unowned self, unowned enterNameVC] name in
                 let success = OwnerKeyController.importKey(connection: connection, wallet: wallet, name: name)
 
-                if !success {
+                if success {
```

Comment:
to prevent code duplication this part could be extracted and placed in a passcode related handler/controller

Reply:
I think we should have a refactoring all related onboarding stuff in a separate issue. In every such screen is completion blocks and repetition. I have untangled some in my PR #2218 but I would do a separate issue to do this for all other places.